File: sql-insight-engine/sql-insight-engine-innodb/src/main/java/org/gongxuanzhang/sql/insight/core/engine/storage/innodb/factory/RecordHeaderFactory.java
package org.gongxuanzhang.sql.insight.core.engine.storage.innodb.factory;

import org.gongxuanzhang.sql.insight.core.engine.storage.innodb.page.ConstantSize;
import org.gongxuanzhang.sql.insight.core.engine.storage.innodb.page.compact.RecordHeader;
import org.gongxuanzhang.sql.insight.core.engine.storage.innodb.page.compact.RecordType;

/**
 * @author gongxuanzhangmelt@gmail.com
 **/
public abstract class RecordHeaderFactory {

    public static RecordHeader indexHeader() {
        RecordHeader recordHeader = new RecordHeader();
        recordHeader.setRecordType(RecordType.PAGE);
        recordHeader.setHeapNo(1);
        recordHeader.setDelete(false);
        recordHeader.setNOwned(1);
        recordHeader.setNextRecordOffset(0);
        return recordHeader;
    }

    public static RecordHeader infimumHeader() {
        RecordHeader recordHeader = new RecordHeader();
        recordHeader.setRecordType(RecordType.INFIMUM);
        recordHeader.setHeapNo(1);
        recordHeader.setDelete(false);
        recordHeader.setNOwned(1);
        recordHeader.setNextRecordOffset(ConstantSize.INFIMUM.size());
        return recordHeader;
    }

    public static RecordHeader supremumHeader() {
        RecordHeader recordHeader = new RecordHeader();
        recordHeader.setRecordType(RecordType.SUPREMUM);
        recordHeader.setHeapNo(1);
        recordHeader.setDelete(false);
        recordHeader.setNOwned(1);
        recordHeader.setNextRecordOffset(0);
        return recordHeader;
    }
}

File: sql-insight-engine/sql-insight-engine-innodb/src/main/kotlin/tech/insight/engine/innodb/factory/RecordHeaderFactory.kt
package tech.insight.engine.innodb.factory

import tech.insight.engine.innodb.page.ConstantSize
import tech.insight.engine.innodb.page.compact.RecordHeader
import tech.insight.engine.innodb.page.compact.RecordType

/**
 * @author gongxuanzhangmelt@gmail.com
 */
object RecordHeaderFactory {
    fun indexHeader(): RecordHeader {
        val recordHeader = RecordHeader()
        recordHeader.setRecordType(RecordType.PAGE)
        recordHeader.setHeapNo(1)
        recordHeader.setDelete(false)
        recordHeader.setNOwned(1)
        recordHeader.setNextRecordOffset(0)
        return recordHeader
    }

    fun infimumHeader(): RecordHeader {
        val recordHeader = RecordHeader()
        recordHeader.setRecordType(RecordType.INFIMUM)
        recordHeader.setHeapNo(1)
        recordHeader.setDelete(false)
        recordHeader.setNOwned(1)
        recordHeader.setNextRecordOffset(ConstantSize.INFIMUM.size())
        return recordHeader
    }

    fun supremumHeader(): RecordHeader {
        val recordHeader = RecordHeader()
        recordHeader.setRecordType(RecordType.SUPREMUM)
        recordHeader.setHeapNo(1)
        recordHeader.setDelete(false)
        recordHeader.setNOwned(1)
        recordHeader.setNextRecordOffset(0)
        return recordHeader
    }
}
